test/libsolidity/hooks/LogsHook.cpp

```diff
@@ -0,0 +1,217 @@
+#include <test/libsolidity/hooks/LogsHook.h>
+#include <test/libsolidity/SemanticTest.h>
+#include <test/libsolidity/SolidityExecutionFramework.h>
+#include <test/libsolidity/util/SoltestErrors.h>
+#include <libsolutil/AnsiColorized.h>
+#include <libsolutil/CommonData.h>
+#include <libsolutil/FixedHash.h>
+
+using namespace std;
+using namespace solidity::util;
+
+namespace solidity::frontend::test
+{
+
+LogsHook::LogsHook(SemanticTest* _test): m_test(_test) {
+	using namespace std::placeholders;
+
+	m_executionFramework = dynamic_cast<SolidityExecutionFramework*>(m_test);
+	soltestAssert(m_executionFramework != nullptr, "");
+
+	m_test->addBuiltin("logs.numLogs", std::bind(&LogsHook::numLogs, this, _1));
```

Comment:
Do we really need these as soon as we have forced output for every single event?

Reply:
Not sure what you mean. Theoretically it is possible to remove all `logs.*` builtins, except `logs.expectEvent`. Do you mean this? I thought it would make sense to also have these "low-level" log builtins, maybe just to have them.  But I think they are not really needed.